### DISCUSSION TITLE
Calculating servo torque required for rocket canard control

### DISCUSSION THREAD

**Discussion by SupSonAlv** [GuidodiPasquo/Actively-Stabilized-Model-Rocket-Simulator#9] [General]
Among the servo specifications in a designed rocket, I did not see a field to specify the torque of the servo, would it not work for correct results or specifying which servo torque is required?

There is a lot of effort in this created application. I think it is necessary to advance it and give it its due where it should be.

**Comment by GuidodiPasquo** [comment]
Hello!

It would be very hard to calculate precisely the torque over the fin. In addition, for the scale of the rockets of the simulator, if the servo's placement is not too far off from the aerodynamic center of the fin, the torque shouldn't be something a normal servo couldn't handle.

**Reply by SupSonAlv** [reply]
Hello;
In the winglets I use, I usually put the servo close to the aerodynamic center. However, it may still be good to add parameters like this to test any design well.If the rocket is flying fast, the fins need to be small, yes, which means less torque is required.
However, it is always safer to know the torque required for the fin used. The rocket should not cause any problems in the sky.

![Untitled](https://github.com/GuidodiPasquo/AeroVECTOR/assets/138517098/7c609c17-27d8-41c7-8adc-4daeb73829bc)
I think it calculates the torque required for the servo where I show it in the picture below, right?
-Why there speed is only for 100m/s.

In the meantime, do you plan to further this simulation program?

**Reply by GuidodiPasquo** [reply]
Hello!

No, the data shown there corresponds to the rocket, so, force and moment about the CG. The speed is limited to 100 m/s for no particular reason, just to make more usable the slider, you can change the maximum value in the source code very easily (in src\gui\gui_functions.py -> line 1298)

For the time being I have no plans to continue the development of the sim either.

**Reply by SupSonAlv** [reply]
Thanks for the explanation. When setting a value above "subsonic" in this part of the code. Will it cause problems and miscalculations in practice?

Making some improvements in this simulation can make sense of your effort and I think it makes a difference compared to some applications in existing applications.

**Reply by GuidodiPasquo** [reply]
Yes, setting supersonic velocities will break the program

**Comment by SupSonAlv** [comment]
I haven't seen any explanation about it.
In the 'Set rocket body' tab
-Actuator deflection
-Is 'Force app point' stability(cal) ?
-What is 'time'
Can you explain this part a little more?

**Reply by GuidodiPasquo** [reply]
The Set Rocket Body tab has four sliders to simulate the forces on the rocket and test changes in the configuration before simulating. For example, if you see in the simulation that your rocket looses control at 10º angle of attack, with an actuator deflection of 20º, at 3 seconds from liftoff, you can put those parameters in the sliders in the Set Rocket Body tab and make changes to make sure you have the net moment that you need (i.e. increase the control fin size to get a restaurative moment).

- Actuator deflection: The current deflection of the actuator
- Force app point: The reduction point of all the forces in the rocket, it changes color to green if the force is applied in the opposite direction.
- Time: Takes the thrust of the motor from the thrust vs time curve for the time indicated in the slider.

**Reply by SupSonAlv** [reply]
It means "actuator deflection", meaning "fin cant". This means it's the angle of the fin against the air. is that right ?
By the way, where is the overall weight of the rocket indicated?

**Reply by GuidodiPasquo** [reply]
It is the fin cant, which is the angle between the rocket and the fin. The angle of the fin against the air depends on the angle of attack of the rocket too.

The weight is indicated in the Parameters tab (as liftoff and burnout mass)
